overwatch/src/services/relay.rs

```diff
@@ -172,61 +140,6 @@ where
     }
 }
 
-impl<Service> Relay<Service>
-where
-    Service: ServiceData,
-    Service::Message: 'static,
-{
-    #[must_use]
-    pub const fn new(overwatch_handle: OverwatchHandle) -> Self {
-        Self {
-            overwatch_handle,
-            _bound: PhantomBound {
-                _inner: PhantomData,
-            },
-        }
-    }
-
-    #[cfg_attr(feature = "instrumentation", instrument(skip(self), err(Debug)))]
-    /// Consumes the relay to generate a channel with the requested service.
-    ///
-    /// # Errors
-    ///
-    /// If the service responds with an unexpected message.
-    pub async fn connect(self) -> Result<OutboundRelay<Service::Message>, RelayError> {
-        let (reply, receiver) = oneshot::channel();
-        self.request_relay(reply).await;
-        self.handle_relay_response(receiver).await
-    }
```

Comment:
Yeah. One of the reasons to have the connect was to be able to keep a relay before services run. Also to keep it so it could be called without an overwatch handle afterward. But in reality is not needed. Nice!

Reply:
Yeah I saw it used in a few places, but in most services we would store the service handle anyway since it's (rightfully) provided in the `ServiceCore::init` method, so they would always have it available. The handle can be cloned and passed around, so I think it's not a big deal.